shared/settings/advanced/index.tsx

```diff
@@ -208,6 +249,153 @@ class Developer extends React.Component<Props, State> {
   }
 }
 
+// A list so the order of the elements is fixed
+const ProxyTypeToDisplayNameList = [
+  ['noProxy', 'No Proxy'],
+  ['httpConnect', 'HTTP Connect'],
+  ['socks', 'Socks5'],
+]
+const ProxyTypeToDisplayName = ProxyTypeToDisplayNameList.reduce((obj, item) => {
+  obj[item[0]] = item[1]
+  return obj
+}, {})
+const DisplayNameToProxyType = {
+  'HTTP Connect': 'httpConnect',
+  'No Proxy': 'noProxy',
+  Socks5: 'socks',
+}
+
+type ProxyState = {
+  address: string
+  port: string
+  proxyType: string
+  certPinning: boolean
+  showCertPinningConfirmationModal: boolean
+}
+
+type ProxyProps = Props & {confirmDisableCertPinning: (disableCertPinning: () => void) => void}
+
+class ProxySettings extends React.Component<ProxyProps, ProxyState> {
+  constructor(props: ProxyProps) {
+    super(props)
+
+    this.state = {
+      address: '',
+      certPinning: true,
+      port: '',
+      proxyType: ProxyTypeToDisplayName['noProxy'],
+      showCertPinningConfirmationModal: false,
+    }
+
+    RPCTypes.configGetProxyDataRpcPromise()
+      .then(this.handleProxyData)
+      .catch(error => console.warn('Error in retrieving proxy data, using default data:', error))
+  }
+
+  handleProxyData = (data: RPCTypes.ProxyData) => {
+    var addressPort = data.addressWithPort.split(':')
+    var address = addressPort[0]
+    var port = '80'
+    if (addressPort.length >= 2) {
+      port = addressPort[1]
+    }
+
+    var certPinning = data.certPinning
+
+    var proxyType = ProxyTypeToDisplayName[RPCTypes.ProxyType[data.proxyType]]
+
+    this.setState({address, certPinning, port, proxyType})
+  }
+
+  handleAddressChange = (address: string) => {
+    this.setState({address})
+  }
+
+  handlePortChange = (port: string) => {
+    this.setState({port})
+  }
+
+  setProxyType = (proxyType: string) => {
+    this.setState({proxyType})
+  }
+
+  toggleCertPinning = () => {
+    if (this.state.certPinning) {
+      this.props.confirmDisableCertPinning(() => this.setState({certPinning: false}))
+    } else {
+      this.setState({certPinning: !this.state.certPinning})
+    }
+  }
+
+  saveProxySettings = () => {
+    var proxyData = {
```

Comment:
This'd be an `onSaveProxySettings` coming from the container that calls dispatch

Reply:
Done.